### ISSUE TITLE
2.12.3: pep517 build and sphinx warnings

### ISSUE THREAD

**Issue by kloczek** [aio-libs/aiobotocore#1109]
**Describe the bug**

**Checklist**
- [x] I have reproduced in environment where `pip check` passes without errors
- [x] I have provided `pip freeze` results
- [x] I have provided sample code or detailed way to reproduce
- [ ] I have tried the same code in botocore to ensure this is an aiobotocore specific issue
- [ ] I have tried similar code in aiohttp to ensure this is is an aiobotocore specific issue
- [ ] I have checked the latest and older versions of aiobotocore/aiohttp/python to see if this is a regression / injection

Looks like pep517 based build shows some warnings
```cosole
+ /usr/bin/python3 -sBm build -w --no-isolation
* Getting build dependencies for wheel...
running egg_info
creating aiobotocore.egg-info
writing aiobotocore.egg-info/PKG-INFO
writing dependency_links to aiobotocore.egg-info/dependency_links.txt
writing requirements to aiobotocore.egg-info/requires.txt
writing top-level names to aiobotocore.egg-info/top_level.txt
writing manifest file 'aiobotocore.egg-info/SOURCES.txt'
[04/15/24 18:13:25] ERROR    listing git files failed - pretending there aren't any                                                                                                 git.py:24
reading manifest file 'aiobotocore.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no previously-included files matching '*.pyc' found anywhere in distribution
warning: no previously-included files matching '*.swp' found anywhere in distribution
adding license file 'LICENSE'
writing manifest file 'aiobotocore.egg-info/SOURCES.txt'
* Building wheel...
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/aiobotocore
copying aiobotocore/__init__.py -> build/lib/aiobotocore
copying aiobotocore/_endpoint_helpers.py -> build/lib/aiobotocore
copying aiobotocore/_helpers.py -> build/lib/aiobotocore
copying aiobotocore/args.py -> build/lib/aiobotocore
copying aiobotocore/awsrequest.py -> build/lib/aiobotocore
copying aiobotocore/client.py -> build/lib/aiobotocore
copying aiobotocore/config.py -> build/lib/aiobotocore
copying aiobotocore/configprovider.py -> build/lib/aiobotocore
copying aiobotocore/credentials.py -> build/lib/aiobotocore
copying aiobotocore/discovery.py -> build/lib/aiobotocore
copying aiobotocore/endpoint.py -> build/lib/aiobotocore
copying aiobotocore/eventstream.py -> build/lib/aiobotocore
copying aiobotocore/handlers.py -> build/lib/aiobotocore
copying aiobotocore/hooks.py -> build/lib/aiobotocore
copying aiobotocore/httpchecksum.py -> build/lib/aiobotocore
copying aiobotocore/httpsession.py -> build/lib/aiobotocore
copying aiobotocore/paginate.py -> build/lib/aiobotocore
copying aiobotocore/parsers.py -> build/lib/aiobotocore
copying aiobotocore/regions.py -> build/lib/aiobotocore
copying aiobotocore/response.py -> build/lib/aiobotocore
copying aiobotocore/retryhandler.py -> build/lib/aiobotocore
copying aiobotocore/session.py -> build/lib/aiobotocore
copying aiobotocore/signers.py -> build/lib/aiobotocore
copying aiobotocore/stub.py -> build/lib/aiobotocore
copying aiobotocore/tokens.py -> build/lib/aiobotocore
copying aiobotocore/utils.py -> build/lib/aiobotocore
copying aiobotocore/waiter.py -> build/lib/aiobotocore
running egg_info
writing aiobotocore.egg-info/PKG-INFO
writing dependency_links to aiobotocore.egg-info/dependency_links.txt
writing requirements to aiobotocore.egg-info/requires.txt
writing top-level names to aiobotocore.egg-info/top_level.txt
[04/15/24 18:13:25] ERROR    listing git files failed - pretending there aren't any                                                                                                 git.py:24
reading manifest file 'aiobotocore.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no previously-included files matching '*.pyc' found anywhere in distribution
warning: no previously-included files matching '*.swp' found anywhere in distribution
adding license file 'LICENSE'
writing manifest file 'aiobotocore.egg-info/SOURCES.txt'
/usr/lib/python3.10/site-packages/setuptools/command/build_py.py:207: _Warning: Package 'aiobotocore.retries' is absent from the `packages` configuration.
!!

        ********************************************************************************
        ############################
        # Package would be ignored #
        ############################
        Python recognizes 'aiobotocore.retries' as an importable package[^1],
        but it is absent from setuptools' `packages` configuration.

        This leads to an ambiguous overall configuration. If you want to distribute this
        package, please make sure that 'aiobotocore.retries' is explicitly added
        to the `packages` configuration field.

        Alternatively, you can also rely on setuptools' discovery methods
        (for example by using `find_namespace_packages(...)`/`find_namespace:`
        instead of `find_packages(...)`/`find:`).

        You can read more about "package discovery" on setuptools documentation page:

        - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html

        If you don't want 'aiobotocore.retries' to be distributed and are
        already explicitly excluding 'aiobotocore.retries' via
        `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,
        you can try to use `exclude_package_data`, or `include-package-data=False` in
        combination with a more fine grained `package-data` configuration.

        You can read more about "package data files" on setuptools documentation page:

        - https://setuptools.pypa.io/en/latest/userguide/datafiles.html


        [^1]: For Python, any directory (with suitable naming) can be imported,
              even if it does not contain any `.py` files.
              On the other hand, currently there is no concept of package data
              directory, all directories are treated like packages.
        ********************************************************************************

!!
  check.warn(importable)
creating build/lib/aiobotocore/retries

[..]

removing build/bdist.linux-x86_64/wheel
Successfully built aiobotocore-2.12.3-py3-none-any.whl
```
And shpinx as well show some future deprecation warning
```console
+ /usr/bin/sphinx-build -n -T -b man docs build/sphinx/man
Running Sphinx v7.2.6
making output directory... done
WARNING: The pre-Sphinx 1.0 'intersphinx_mapping' format is deprecated and will be removed in Sphinx 8. Update to the current format as described in the documentation. Hint: "intersphinx_mapping = {'<name>': ('https://docs.python.org/', None)}".https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#confval-intersphinx_mapping
loading intersphinx inventory from https://docs.python.org/objects.inv...
intersphinx inventory has moved: https://docs.python.org/objects.inv -> https://docs.python.org/3/objects.inv
building [mo]: targets for 0 po files that are out of date
writing output...
building [man]: all manpages
updating environment: [new config] 9 added, 0 changed, 0 removed
reading sources... [100%] tutorial
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... python-aiobotocore.3 { tutorial examples examples/s3/basic_usage examples/sqs/queue examples/sqs/producer_consumer examples/dynamo_db/insertions examples/dynamo_db/tables contributing } done
build succeeded, 1 warning.
```

**Comment by jakob-keller** [comment]
Thanks for reporting. This might have been fixed by #1129. Could you please check again?

**Comment by jakob-keller** [comment]
The sphinx issue was fixed in #1155
